bc_obps/service/contact_service.py

```diff
@@ -37,6 +40,34 @@ def get_if_authorized(cls, user_guid: UUID, contact_id: int) -> Optional[Contact
             raise Exception(UNAUTHORIZED_MESSAGE)
         return contact
 
+    @classmethod
+    def get_with_places_assigned(cls, user_guid: UUID, contact_id: int) -> Optional[Contact]:
+        contact = cls.get_if_authorized(user_guid, contact_id)
+
+        places_assigned = []
+        if contact:
+            for contact_field in contact._meta.get_fields():
+                if isinstance(contact_field, (ManyToOneRel, ManyToManyRel)):
+                    related_objects = getattr(contact, contact_field.name).all()
+
+                    for object in related_objects:
+                        role_name = contact.business_role.role_name
+                        if isinstance(object, Operator):
+                            place = f"{role_name} - {object.legal_name}"
+                        elif isinstance(object, Operation):
+                            place = f"{role_name} - {object.name}"
+                        else:
+                            # If the relation doesn't have a custom format (e.g. for operator we want to show the legal name), then we just show the contact role and the related model name
+                            place = f"{role_name} - {related_objects.model.__name__}"
```

Comment:
recommendation: split this string formatting logic out into a separate (private) helper function

Reply:
What do you mean by private?

Reply:
just prefixing the name of the function with `_`